libpod/image/image.go

```diff
@@ -1487,14 +1487,14 @@ func (i *Image) Save(ctx context.Context, source, format, output string, moreTag
 		}
 		manifestType = manifest.DockerV2Schema2MediaType
 	case "docker-archive", "":
-		dst := output
 		destImageName := imageNameForSaveDestination(i, source)
-		if destImageName != "" {
-			dst = fmt.Sprintf("%s:%s", dst, destImageName)
+		ref, err := dockerArchiveDstReference(destImageName)
+		if err != nil {
+			return err
 		}
-		destRef, err = dockerarchive.ParseReference(dst) // FIXME? Add dockerarchive.NewReference
+		destRef, err = dockerarchive.NewReference(output, ref)
 		if err != nil {
-			return errors.Wrapf(err, "error getting Docker archive ImageReference for %q", dst)
+			return errors.Wrapf(err, "error getting Docker archive ImageReference for %q", output)
```

Comment:
It might be useful to include both `output` and `ref.String()` here.

Reply:
Included.